app/process/watcher/transfer/watcher.go

```diff
@@ -136,16 +141,17 @@ func (ctw Watcher) beginWatching(q *pair.Queue) {
 	}
 }
 
-func (ctw Watcher) processTransaction(tx mirror_node.Transaction, q *pair.Queue) {
+func (ctw Watcher) processTransaction(tx mirror_node.Transaction, q qi.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
 	amount, nativeAsset, err := tx.GetIncomingTransfer(ctw.accountID.String())
 	if err != nil {
 		ctw.logger.Errorf("[%s] - Could not extract incoming transfer. Error: [%s]", tx.TransactionID, err)
 		return
 	}
 
-	wrappedAsset, err := ctw.contractService.ToWrapped(nativeAsset)
-	if err != nil {
+	// TODO: Figure indexing out, for now we are simply looking at mapping of Hedera native asset to Ethereum wrapped asset
```

Comment:
You can get the `targetChain` from the `memo` field. By definition the memo will be: `{targetChain}-{receiverAddress}`

Reply:
Yes. I have it in my notes for the very next PR, related to the Hedera -> EVM flow.